vmdb/app/models/ems_kubernetes.rb

```diff
@@ -36,27 +52,40 @@ def api_endpoint
     self.class.raw_api_endpoint(hostname, port)
   end
 
-  def connect(_options = {})
-    self.class.raw_connect(hostname, port)
-  end
+  def connect(options = {})
+    hostname = options[:hostname] || self.address
+    port     = options[:port]     || self.port
 
-  def self.event_monitor_class
-    MiqEventCatcherKubernetes
+    self.class.raw_connect(hostname, port)
   end
 
-  def authentication_check
+  def verify_credentials(auth_type = nil, options = {})
     # TODO: support real authentication using certificates
-    [true, ""]
-  end
+    options = options.merge(:auth_type => auth_type)
 
-  def verify_credentials(_auth_type = nil, _options = {})
-    # TODO: support real authentication using certificates
-    true
+    with_provider_connection(options) do |connection|
+      unless (connection.api['versions'] || []).include? api_version
```

Comment:
The reason I think it's kubeclient's responsibility is that it's passed as a parameter to Kubeclient.new.  Therefore we are telling kubeclient which version we need, and it's own validator should verify that the version is available and accessible.  Thus, if we hypothetically connected to a kubernetes instance that did not have the version we gave to kubeclient, then kubeclient should tell us that (ideally on the api_valid? method).  If we were not passing the api_version in, then I'd agree with you.